dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java

```diff
@@ -0,0 +1,172 @@
+                .map( period -> toSqlCondition( period, timeField.get() ) )
+                .collect( Collectors.joining( " or ", "(", ")" ) ) );
+        }
+        else
+        {
+            sql.append( quote( ANALYTICS_TBL_ALIAS, params.getPeriodType().toLowerCase() )
+                + " in ("
+                + getQuotedCommaDelimitedString( getUids( params.getDimensionOrFilterItems( PERIOD_DIM_ID ) ) )
+                + ") " );
+        }
+        return sql.toString();
+    }
+
+    @Override
+    protected String getSqlConditionHasStartEndDate( EventQueryParams params )
+    {
+        TimeField timeField = getTimeField( params ).orElse( TimeField.ENROLLMENT_DATE );
+        return timeField.getField() + " >= '" + getMediumDateString( params.getStartDate() ) + "' and "
+            + timeField.getField() + " <= '"
+            + getMediumDateString( params.getEndDate() ) + "' ";
+    }
+
+    @Override
+    protected String getSqlConditionForNonDefaultBoundaries( EventQueryParams params )
+    {
+
+        String sql = params.getProgramIndicator().getAnalyticsPeriodBoundaries().stream()
+            .filter(
+                boundary -> boundary.isCohortDateBoundary() && !boundary.isEnrollmentHavingEventDateCohortBoundary() )
+            .map( boundary -> statementBuilder.getBoundaryCondition( boundary, params.getProgramIndicator(),
+                params.getTimeFieldAsField(),
+                params.getEarliestStartDate(), params.getLatestEndDate() ) )
+            .collect( Collectors.joining( " and " ) );
+
+        String sqlEventCohortBoundary = params.getProgramIndicator().hasEventDateCohortBoundary()
+            ? getProgramIndicatorEventInProgramStageSql( params.getProgramIndicator(), params.getEarliestStartDate(),
+                params.getLatestEndDate() )
+            : "";
+
+        return Stream.of( sql, sqlEventCohortBoundary )
+            .filter( StringUtils::isNotBlank )
+            .collect( Collectors.joining( " and " ) );
+    }
+
+    private String getProgramIndicatorEventInProgramStageSql( ProgramIndicator programIndicator,
+        Date reportingStartDate, Date reportingEndDate )
+    {
+        Assert.isTrue( programIndicator.hasEventDateCohortBoundary(),
+            "Can not get event date cohort boundaries for program indicator:" + programIndicator.getUid() );
+
+        Map<String, Set<AnalyticsPeriodBoundary>> map = programIndicator.getEventDateCohortBoundaryByProgramStage();
+
+        final SimpleDateFormat format = new SimpleDateFormat();
+        format.applyPattern( Period.DEFAULT_DATE_FORMAT );
+
+        String sql = "";
+        for ( String programStage : map.keySet() )
+        {
+            Set<AnalyticsPeriodBoundary> boundaries = map.get( programStage );
+
+            String eventTableName = "analytics_event_" + programIndicator.getProgram().getUid();
+            sql += " (select count(*) from " + eventTableName + " where " + eventTableName +
```

Comment:
waht do you think about  replacement of  '... ( select count(*)..) > 0' with 'exists(select 1 from ... )'

Reply:
makes sense and should improve performances. So generally speaking I agree with that.
However this code was just moved from somewhere else and is not meant to be modified within this ticket scope. 
I'd suggest to create a specific task for it

Reply:
Ok, agree